test/e2e/service.go

```diff
@@ -180,4 +181,131 @@ var _ = Describe("Services", func() {
 		Expect(foundRW).To(Equal(true))
 		Expect(foundRO).To(Equal(true))
 	})
+
+	It("should serve basic a endpoint from pods", func(done Done) {
+		serviceName := "endpoint-test"
+		ns := api.NamespaceDefault
+		labels := map[string]string{
+			"foo": "bar",
+			"baz": "blah",
+		}
+
+		defer func() {
+			err := c.Services(ns).Delete(serviceName)
```

Comment:
According to the docs, it's only needed in a goroutine.

http://onsi.github.io/ginkgo/#marking-specs-as-failed

Does a `defer` block constitue a goroutine? I thought it was run in the main thread... I think it should be possible to write an experiment with a failure inside the `defer` block to see whether Ginkgo is actually catching the error...